### DISCUSSION TITLE
Feature Request: Keep image names

### DISCUSSION THREAD

**Discussion by enesaltinkaya** [zeux/meshoptimizer#682] [General]
Hi,

Blender has a new features regarding images while exporting GLTF/GLB;
- export_unused_images (boolean, (optional)) – Unused images, Export images not assigned to any material

When using this feature, one needs to know names to be able to distinguish between images.
Similar to **-kn** can we have a **-kin** "keep image names" for example?

The second image below named "test_Albedo" is not connected to any materials. Both images are packed in .blend file.
This could be useful when one wants to carry over additinonal images with glb, like heightmaps/bump maps.

And once again thanks a lot for this awesome library!

before
<img src="https://github.com/zeux/meshoptimizer/assets/8594559/3fe2eaff-9104-4ecb-9031-5cbc99bffc3a" width="200">

after
<img src="https://github.com/zeux/meshoptimizer/assets/8594559/59b61406-0f6f-4045-a7e4-4516de2969c9" width="200">

**Comment by zeux** [comment]
Just to double check, are you interested in *both* keeping the names and keeping the unreferenced named images? The former I can just do unconditionally if the names are useful for your application. The latter would indeed require a separate option.

**Reply by enesaltinkaya** [reply]
Both would be ideal for my usecase.

What I'm doing right now is I'm creating "image holder objects" and assign images to color, roughness, normal nodes to it's material just so i can access them in the code. It gets complicated and cumbersome.

**Reply by zeux** [reply]
Ok that makes sense, I opened #683 to track this.

**Reply by enesaltinkaya** [reply]
Alright, thanks!

**Reply by zeux** [reply]
Actually, gltfpack doesn’t remove unreferenced images right now. So maybe just preserving the names is sufficient? Even in the example above both images are kept, just the names are removed. I will make the change to unconditionally keep the names for now.

**Reply by enesaltinkaya** [reply]
You are the best! :)

Yep all i need is the name field on the json.

**Reply by zeux** [reply]
Should be fixed in latest, if you need a binary build you can grab it from artifacts on this page https://github.com/zeux/meshoptimizer/actions/runs/8904081011

**Reply by enesaltinkaya** [reply]
Brilliant, thanks a lot mate.

Now i can export splat maps, height maps, erosion maps without the extra material hassle :)
